source/computer/commands.rst

```diff
@@ -63,54 +62,489 @@ Linux/macOS 下有成百上千个命令，每个命令都有众多选项。这
+
+    # 以上命令只是将转换后的文本内容打印出来，并为改变文件本身。可以使用 -i 选项直接改变文件
+    $ sed -i 's/book/books/g' file
+
+    # 以上命令使用斜杠 / 当定界符，也可以使用任意定界符
+    $ sed 's#book#books#' file
+    $ sed 's#book#books#g' file
+    $ sed -i 's#book#books#g' file
+
 sort
 ----
 
+``sort`` 命令可以将文件内容进行排序，并打印排序结果。该命令将文件的每一行作为一个单位，相互比较。
+默认的比较原则是从首字符向后，依次按 ASCII 码值进行比较，最后将他们按排序结果输出。
+
+我们使用示例文件 :file:`seismo-learn-sort.txt` 展示该命令的主要用法::
+
+    $ cat seismo-learn-sort.txt
+    6:software:seisman:-1.3
+    1:seismology101:zhaozhiyuan1989:291
+    7:software:core-man:101.1
+    1:seismology101:zhaozhiyuan1989:291
+    2:seismology101:seisman:80
+    3:seismology101:wangliang1989:101.2
+    8:software:zhaozhiyuan1989:291
+    5:seismology:core-man:-81.2
+    4:seismology:seisman:91
+    1:seismology101:zhaozhiyuan1989:291
+
+::
+
+    # 按 ASCII 码值进行升序排序
+    $ sort seismo-learn-sort.txt
+
+    # 按 ASCII 码值进行降序排序
+    $ sort -r seismo-learn-sort.txt
+
+    # 按 ASCII 码值进行升序排序，并忽略相同行（即重复行只统计一次）
+    $ sort -u seismo-learn-sort.txt
+
+    # 按 ASCII 码值进行降序排序，并忽略相同行
+    $ sort -u -r seismo-learn-sort.txt
+    # 不同选项也可以写在一块
+    # sort -ur seismo-learn-sort.txt
+
+    # 按数值大小进行升序排序
+    $ sort -n seismo-learn-sort.txt
+
 ssh
 ---
 
+``ssh`` 命令的命名源于 **S**\ ecure **Sh**\ ell（安全外壳协议，简称 SSH），该协议是
+一种加密的网络传输协议。使用 ``ssh`` 命令可以登录到远程计算机中。常用于登录服务器提交计算任务。
+
+若远程计算机的 IP 地址是 192.168.1.100，用户名是 seismo-learn::
+
+    # 登录到远程计算机
+    $ ssh seismo-learn@192.168.1.100
+
+    # 以图形界面连接远程计算机（需要配置远程计算上的 ssh 服务器配置）
+    $ ssh -X seismo-learn@192.168.1.100
+
 tail
 ----
 
+``tail`` 命令用于打印文件尾部内容::
+
+    # 打印 /etc/passwd 的后 10 行
+    $ tail /etc/passwd
+
+    # 打印 /etc/passwd 的后 5 行
+    $ tail -n 5 /etc/passwd
+
+如果 ``-n`` 选项后的数字（N）前面有加号 :kbd:`+`，则打印的是文件开头的第 N 行到文件末尾，
+并不是最后 N 行。例如，打印 :file:`/etc/passwd` 的第 5 行到文件末尾::
+
+    $ tail -n +5 /etc/passwd
+
 tar
 ---
 
+``tar`` 命令的名字来自 **t**\ ape **ar**\ chive（磁带存档），因为该命令最初被用来在磁带上
+创建档案。该命令可以把一大堆文件和目录打包成一个文件，并且可以选择压缩该文件，这对于备份文件或
+将几个文件组合成一个文件以便于网络传输是非常有用的。
+
+首先要弄清两个概念：打包和压缩。打包是指将一大堆文件或目录打包成一个文件，而压缩则是将一个大文件
+通过一些压缩算法变成一个小文件。Linux 中的很多压缩程序只能对压缩单个文件，若想压缩一大堆文件，
+首先得将这一大堆文件打成一个包（使用 ``tar`` 命令），再用压缩程序进行压缩
+（使用 ``gzip`` 或 ``bzip2`` 命令）。使用 ``tar`` 命令时，可以直接选择压缩打包的文件，无需
+再单独使用压缩程序进行压缩。
+
+::
+
+    # 将 file1 和 file2 打包，打包文件命名为 seismo-learn.tar
+    # 选项 -f 后的打包文件名一般用 .tar 来作为标识
+    $ tar -cvf seismo-learn.tar file1 file2
+
+    # 同上，但在打包后用 gzip 命令进行压缩
+    # 一般用 .tar.gz 或 .tgz 来作标识
+    $ tar -zcvf seismo-learn.tar.gz file1 file2
+
+    # 打包后，用 bzip2 命令进行压缩
+    # 一般用 .tar.bz2 或 .tbz 来作标识
+    $ tar -jcvf seism-learn.tar file1 file2
+
+查阅打包压缩文件含有哪些文件和目录::
+
+    $ tar -tvf seismo-learn.tar
+    $ tar -tvf seismo-learn.tar.gz
+    $ tar -tvf seismo-learn.tar.tb2
+
+将打包压缩文件还原::
+
+    # 默认还原到当前目录下
+    $ tar -xvf seismo-learn.tar
+    $ tar -xvf seismo-learn.tar.gz
+    $ tar -xvf seismo-learn.tar.bz2
+
+    # 还原到 bak 目录下（该目录必须存在）
+    $ mkdir bak
+    $ tar -xvf seismo-learn.tar -C bak
+    $ tar -xvf seismo-learn.tar.gz -C bak
+    $ tar -xvf seismo-learn.tar.bz2 -C bak
```

Comment:
These commands work for `.tar`, `.tar.gz` and `.tar.bz2`. Don't repeat these commands. Just give commands for `tar.gz` and say that the commands also work for `tar` and `tar.bz2`.

Reply:
done in 4bfcb59.